Marlin/temperature.cpp

```diff
@@ -1390,8 +1395,91 @@ void Temperature::set_current_temp_raw() {
   #endif
   current_temperature_bed_raw = raw_temp_bed_value;
   temp_meas_ready = true;
+
 }
 
+#if ENABLED(PINS_DEBUGGING)
+
+  void endstop_monitor() {
+  /**  
+   * monitors endstops & Z probe for changes
+   *
+   * If a change is detected then the LED is toggled and
+   * a message is sent out the serial port
+   *
+   * Yes, we could miss a rapid back & forth change but
+   * that won't matter because this is all manual.
+   *
+   */ 
```

Comment:
Comments like this should be placed above the function definition rather than within it.